### PR TITLE
Add a code style guide to help Rust beginners

### PR DESCRIPTION
**Pull request by KSXGitHub**
This guide will hopefully reduce the number of code reviews and change requests in the future, speeding up the merging process.

### CODE REVIEW

**Comment by KSXGitHub** [comment]
@pnpm/rust There are more that can be added here from the code reviews. If you want to add them, feel free.

**Comment by KSXGitHub** [comment]
@zkochan Can you review this?

**Review by zkochan — CHANGES_REQUESTED** [review]

File: CODE_STYLE_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+let a = node_bin_dir(&my_path_buf);
+let b = node_bin_dir(&my_path_ref.to_path_buf());
+```
+
+The above code is suboptimal because it forces the [copying] of `my_path_ref` only to be used as a reference.
+
+Changing the signature of `workspace` to `&Path` would help remove `.to_path_buf()`, eliminating the unnecessary copying:
+
+```rust
+fn node_bin_dir(workspace: &PathBuf) -> PathBuf {
```
[the remark this answers is not in the document]
Comment (KSXGitHub) [reply CODE_STYLE_GUIDE.md:121]: Thanks for catching that!

[the remark this answers is not in the document]
Comment (KSXGitHub) [reply CODE_STYLE_GUIDE.md:121]:
```suggestion
fn node_bin_dir(workspace: &Path) -> PathBuf {
```

[the remark this answers is not in the document]
Comment (KSXGitHub) [reply CODE_STYLE_GUIDE.md]: Fixed!

**Review by zkochan — APPROVED** [review]

**Pull request merged**